src/aws/aws_sdk/auth.rs

```diff
@@ -0,0 +1,59 @@
+use crate::aws::auth::AwsAuthentication;
+use aws_config::default_provider::credentials::default_provider;
+use aws_config::meta::credentials::LazyCachingCredentialsProvider;
+use aws_config::profile::ProfileFileCredentialsProvider;
+
+use aws_config::sts::AssumeRoleProviderBuilder;
+
+use aws_types::credentials::SharedCredentialsProvider;
+
+use aws_types::Credentials;
+
+impl AwsAuthentication {
+    pub async fn credentials_provider(&self) -> SharedCredentialsProvider {
+        match self {
+            Self::Static {
+                access_key_id,
+                secret_access_key,
+            } => SharedCredentialsProvider::new(Credentials::from_keys(
+                access_key_id,
+                secret_access_key,
+                None,
+            )),
+            AwsAuthentication::File {
+                credentials_file,
+                profile,
+            } => {
+                warn!("Overriding the credentials file is not supported. `~/.aws/config` and `~/.aws/credentials` will be used instead of \"{}\"", credentials_file);
```

Comment:
Did we note this in the docs for this source?

Reply:
I wasn't sure of a good place to put this. All the AWS docs are shared, and I think this is a pretty minor edge case that very few would hit.